code/game/machinery/cryo.dm

```diff
@@ -473,11 +473,12 @@ var/global/list/cryo_health_indicator = list(	"full" = image("icon" = 'icons/obj
 	if(usr == occupant)//If the user is inside the tube...
 		if (usr.isDead())//and he's not dead....
 			return
-		to_chat(usr, "<span class='notice'>Release sequence activated. This will take two minutes.</span>")
-		sleep(1200)
+		to_chat(usr, "<span class='notice'>Release sequence activated. This will take thirty seconds.</span>")
+		sleep(300)
 		if(!src || !usr || !occupant || (occupant != usr)) //Check if someone's released/replaced/bombed him already
 			return
 		go_out()//and release him from the eternal prison.
+		occupant.bodytemperature = T0C+25 // because they've suffered enough.

```

Comment:
That is specifically for people who self eject, and was a compromise between the two opinions.